src/index.ts

```diff
@@ -113,23 +114,24 @@ const extension: JupyterFrontEndPlugin<void> = {
       setting.changed.connect(updateSettings);
     }
 
-    let cellStartTime = new Date();
-
     NotebookActions.executionScheduled.connect((_, args) => {
+      const { cell } = args;
       if (enabled) {
-        cellStartTime = new Date();
+        cellExecutionTimeMetadata[cell.model.id] = { startTime: new Date(), endTime: new Date() };
```

Comment:
I tried to use `cell.id` to get the cell's unique identifier, but it looks like it's undefined. I was able to get the id from
`cell.model.id`. I created a bug report on Jupyterlab: https://github.com/jupyterlab/jupyterlab/issues/10697